### PR TITLE
BY add referral code to customer info

### PR DESCRIPTION
**Pull request by Beiancaa**

### CODE REVIEW

File: app/models/customer_info.rb [hunk 1]
```diff
@@ -4,5 +4,25 @@ class CustomerInfo < ApplicationRecord
   validates_length_of :name, :father_name, :grandfather_name, :family_name, maximum: 35
   validates_length_of :name, :father_name, :grandfather_name, :family_name, minimum: 2
   validates :points_balance, numericality: { greater_than_or_equal_to: 0 }
-#   validates_format_of :home_phone_number, with: /\A(?!([1-9]))[0-9]{11}\z/#, message: :bad_home_phone_number
+  #validates_format_of :home_phone_number, with: /\A(?!([1-9]))[0-9]{11}\z/#, message: :bad_home_phone_number
+  before_validation :generate_referral_code, on: :create
+
+  def check_referral_code_existing
```
Comment (ssharba) [inline app/models/customer_info.rb]: I would move this logic to the controller

File: app/views/customer_infos/_form.html.haml [hunk 2]
```diff
@@ -53,10 +53,10 @@
                     %br/
                     = form.text_field :home_phone_number, id: :customer_info_home_phone_number, class:"form-control g-color-black g-bg-white g-bg-white--focus g-brd-gray-light-v4 g-brd-primary--hover rounded g-py-13 g-px-15"
                   %br/
-                  .mp-4
-                    %label.g-color-gray-dark-v2.g-font-weight-700.g-font-size-16= t 'views.customer_info.new.referred_by'
-                    %br/
-                    = form.text_field :referred_by, id: :customer_info_referral_code, class:"form-control g-color-black g-bg-white g-bg-white--focus g-brd-gray-light-v4 g-brd-primary--hover rounded g-py-13 g-px-15"  
+                  -# .mp-4
```
Comment (MwMaki) [inline app/views/customer_infos/_form.html.haml]: @nooralnabi  Please remove any un-wanted code

**Comment by ssharba** [comment]
conflicts

**Comment by nooralnabi** [comment]
@ssharba I fix the conflicts

**Pull request merged**